src/helpers/models.rs

```diff
@@ -0,0 +1,140 @@
+    event: Event,
+
+    /// A key to group sets of the events into.
+    breakdown_key: PlainText,
+}
+
+#[cfg(feature = "debug")]
+impl Debug for SourceEvent {
+    fn fmt(&self, f: &mut Formatter<'_>) -> core::fmt::Result {
+        write!(
+            f,
+            "SourceEvent:\n  matchkeys={:?}\n  breakdown_key={}",
+            &self.event.matchkeys, &self.breakdown_key
+        )
+    }
+}
+
+#[cfg_attr(feature = "enable-serde", derive(Serialize, Deserialize))]
+struct TriggerEvent {
+    event: Event,
+
+    /// Conversion value.
+    value: SecretShare,
+
+    /// Zero knowledge proof that the trigger value lies within a specific range
+    /// of values. The range is specified in [TriggerFanoutQuery].
+    zkp: PlainText,
+}
+
+#[cfg(feature = "debug")]
+impl Debug for TriggerEvent {
+    fn fmt(&self, f: &mut Formatter<'_>) -> core::fmt::Result {
+        write!(
+            f,
+            "TriggerEvent:\n  matchkeys={:?}\n  value={:?}",
+            &self.event.matchkeys, &self.value
+        )
+    }
+}
+
+#[cfg_attr(feature = "enable-serde", derive(Serialize, Deserialize))]
+enum QueryType {
+    SourceFanout,
+    TriggerFanout,
+}
+
+#[cfg_attr(feature = "enable-serde", derive(Serialize, Deserialize))]
+enum Node {
+    Helper1,
+    Helper2,
+    Helper3,
+}
+
+#[cfg_attr(feature = "enable-serde", derive(Serialize, Deserialize))]
+struct IPAQuery {
+    /// Caller authentication token.
+    auth_token: PlainText,
+
+    /// Initial MPC helper node to send the data to.
+    leader_node: Node,
+
+    /// List of match key providers to be used by the source and trigger events during an epoch.
+    mk_providers: Vec<String>,
+
+    /// Source-fanout or Trigger-fanout.
+    query_type: QueryType,
+
+    /// Percentage of epoch-level privacy budget this query should consume. Likely 1-100.
+    privacy_budget: PBRange,
+
+    /// A collection of source events. At least 100 (TBD) unique source events must be provided.
+    source_events: Vec<SourceEvent>,
+
+    /// A collection of trigger events. At least 10 (TBD) unique trigger events must be provided.
+    trigger_events: Vec<TriggerEvent>,
+}
+
+#[cfg_attr(feature = "enable-serde", derive(Serialize, Deserialize))]
+struct SourceFanoutQuery {
+    query: IPAQuery,
+
+    /// The maximum number of attributed conversion events that a single person can contribute
+    /// towards the final output.
+    cap: u8,
+}
+
+#[cfg(feature = "debug")]
+impl Debug for SourceFanoutQuery {
+    fn fmt(&self, f: &mut Formatter<'_>) -> core::fmt::Result {
+        write!(
+            f,
+            "SourceFanoutQuery:\n  {} source events\n  {} trigger events",
+            &self.query.source_events.len(),
+            &self.query.trigger_events.len()
+        )
+    }
+}
+
+#[cfg_attr(feature = "enable-serde", derive(Serialize, Deserialize))]
+struct TriggerFanoutQuery {
+    query: IPAQuery,
+
+    /// The range which all trigger event's conversion values must lie within.
```

Comment:
nit: the trailing preposition bothers me =).

I suggest: "The range within which all the trigger event values must lie"